lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -318,6 +319,7 @@ def self.resolve_automation_object(uri, user_obj, attr = nil, options = {}, read
     raise "User object not passed in" unless user_obj.kind_of?(User)
     uri = create_automation_object(uri, attr, options) if attr
     options[:uri] = uri
+    User.current_user = user_obj
```

Comment:
@kbrock 
This is done pretty early on as soon as we dequeue, so the only way this code will get executed again is if it got queued again because of a retry. The retry could run on a completely different process or worker. This isn't a recursive call.